DependencyInjection/Compiler/ConfigurationCheckPass.php

```diff
@@ -30,7 +30,8 @@ public function process(ContainerBuilder $container)
         }
 
         if ($container->has('fos_rest.view_response_listener') && isset($container->getParameter('kernel.bundles')['SensioFrameworkExtraBundle'])) {
-            if (!$container->has('sensio_framework_extra.view.listener')) {
+            $config = $container->getExtensionConfig('sensio_framework_extra');
+            if (!(true === $config[0]['view']['annotations'] || $container->has('sensio_framework_extra.view.listener')))  {
```

Comment:
this is wrong for multiple reasons:

- you check only the first config of the bundle, which is not the final config at all (subsequent configs could enable it, or could re-disable it)
- there is no guarantee at all that `$config[0]['view']['annotations']` exists (the user might be configuring other parts of SensioFrameworkExtraBundle
- this leaves out any processing that the bundle could do on the config (which includes the fact of defining a default value)

Reply:
Well in that case the only possible check is to loop through all the configuration and merge it. Default configuration is viewListener enabled in FrameworkExtraBundle...
Meaning the whole view.listener check useless and the sensioframeworkextra bundle dep should be properly written in the documentation?

Reply:
Updated the PR to use a xor check against the new TemplateListener which replaces ViewReponseListener in SensioFrameworkExtraBundle  4+ (i.e. it either has new config or old config)